src/connector/src/parser/protobuf/parser.rs

```diff
@@ -382,62 +236,9 @@ pub fn from_protobuf_value<'a>(
         }
         Value::Message(dyn_msg) => {
             if dyn_msg.descriptor().full_name() == "google.protobuf.Any" {
-                // If the fields are not presented, default value is an empty string
-                if !dyn_msg.has_field_by_name("type_url") || !dyn_msg.has_field_by_name("value") {
-                    borrowed!(JsonbRef::empty_string());
-                }
-
-                // Sanity check
-                debug_assert!(
-                    dyn_msg.has_field_by_name("type_url") && dyn_msg.has_field_by_name("value"),
-                    "`type_url` & `value` must exist in fields of `dyn_msg`"
-                );
-
-                // The message is of type `Any`
-                let (type_url, payload) = extract_any_info(dyn_msg);
-
-                let payload_field_desc = dyn_msg.descriptor().get_field_by_name("value").unwrap();
-
-                let payload = from_protobuf_value(&payload_field_desc, &payload, descriptor_pool)?;
-                let Some(ScalarRefImpl::Bytea(payload)) = payload.to_datum_ref() else {
-                    bail_uncategorized!("expected bytes for dynamic message payload");
-                };
-
-                // Get the corresponding schema from the descriptor pool
-                let msg_desc = descriptor_pool
-                    .get_message_by_name(&type_url)
-                    .ok_or_else(|| {
-                        uncategorized!("message `{type_url}` not found in descriptor pool")
-                    })?;
-
-                let f = msg_desc
-                    .clone()
-                    .fields()
-                    .map(|f| f.name().to_string())
-                    .collect::<Vec<String>>();
-
-                let full_name = msg_desc.clone().full_name().to_string();
-
-                // Decode the payload based on the `msg_desc`
-                let decoded_value = DynamicMessage::decode(msg_desc, payload).unwrap();
-                let decoded_value = from_protobuf_value(
-                    field_desc,
-                    &Value::Message(decoded_value),
-                    descriptor_pool,
-                )?
-                .to_owned_datum()
-                .unwrap();
-
-                // Extract the struct value
-                let ScalarImpl::Struct(v) = decoded_value else {
-                    panic!("Expect ScalarImpl::Struct");
-                };
-
-                ScalarImpl::Jsonb(JsonbVal::from(serde_json::json!(recursive_parse_json(
-                    v.fields(),
-                    Some(f),
-                    Some(full_name),
-                ))))
+                ScalarImpl::Jsonb(JsonbVal::from(
+                    serde_json::to_value(dyn_msg).map_err(AccessError::ProtobufAnyToJson)?,
+                ))
```

Comment:
Does it mean using serde_json to serialize the any type message is good enough? Or we will have some following up improvement PRs to change the return type and value?

Reply:
* It is a formally defined conversion in the Protobuf specification, and the best we can do if we want to return `jsonb`.
* An alternative `jsonb` is to follow PostgreSQL rules instead of Protobuf rules (how to handle bytes, integers larger than MAX_SAFE_INTEGER, etc). To keep things simple there is no plan to provide this flavor until requested.
* Beyond `jsonb` there is currently no other type that can handle its dynamic nature. But maybe we will get more feedback or inspiration when it is actually used by someone.
* Well-Known Types other than `Any` may also worth specialization, for example converting `google.protobuf.Timestamp` to `timestamptz` rather than a `struct`, and `google.protobuf.Value` (unsupported now due to its recursive nature) to `jsonb`. But such usability improvements are less prioritized compared to bug fixes.